File: oolab/src/main/java/agh/ics/oop/model/MapDirection.java
package agh.ics.oop.model;

public enum MapDirection {
    NORTH, EAST, SOUTH, WEST;

    public String toString() {
        return switch (this) {
            case EAST -> "Wschód";
            case WEST -> "Zachód";
            case NORTH -> "Północ";
            case SOUTH -> "Południe";
        };
    }
    public MapDirection next(){
        return MapDirection.values()[(ordinal() + 1) % MapDirection.values().length];
    }
    public MapDirection previous(){
        return MapDirection.values()[(ordinal() - 1 + MapDirection.values().length) % MapDirection.values().length];
    }
    public Vector2d toUnitVector(){
        return switch(this){
            case EAST -> new Vector2d(1, 0);
            case NORTH -> new Vector2d(0,1);
            case WEST -> new Vector2d(-1, 0);
            case SOUTH -> new Vector2d(0, -1);
        };
    }
}

File: oolab_kotlin/src/main/kotlin/MapDirection.kt
package org.example

enum class MapDirection(val vector2d: Vector2d) {
    NORTH(Vector2d(0, 1)),
    EAST(Vector2d(1, 0)),
    SOUTH(Vector2d(0, -1)),
    WEST(Vector2d(-1, 0)), ;

    override fun toString(): String {
        return when(this) {
            NORTH -> "Północ";
            EAST -> "Wschód";
            SOUTH -> "Południe";
            WEST -> "Zachód";
        }
    }
    fun next(): MapDirection {
        return entries[this.ordinal + 1% entries.size];
    }

    fun previous(): MapDirection {
        return entries[(this.ordinal - 1 + entries.size)% entries.size];
    }



}